### PR TITLE
Improve loading experience

### PR DESCRIPTION
**Pull request by nop33**
I expect using the new `/addresses/<address>/tokens-balance` will soon improve the experience even more.

### CODE REVIEW

**Review by mvaivre — COMMENTED** [review]
👏

File: src/components/AddressesTokensList.tsx [hunk 1]
```diff
@@ -29,47 +30,60 @@ import { useAppSelector } from '~/hooks/redux'
...
+
+type CarouselPageEntry = Asset | UnknownTokensEntry | LoadingIndicator
+
 const PAGE_SIZE = 3
 
-const AddressesTokensList = ({ addresses: addressesParam, style }: AddressesTokensListProps) => {
-  const allAddresses = useAppSelector(selectAllAddresses)
-  const addresses = addressesParam ?? allAddresses
-  const addressHashes = addresses.map(({ hash }) => hash)
+const AddressesTokensList = ({ address: addressParam, style }: AddressesTokensListProps) => {
```
[the remark this answers is not in the document]
Comment (nop33) [reply src/components/AddressesTokensList.tsx:54]: Absolutely right. This is an oversight from my side. Initially `address` was of type `Address`. Then I changed the type to `AddressHash` but forgot to rename the var 👍

File: src/components/AddressesTokensList.tsx [hunk 2]
```diff
@@ -177,4 +198,14 @@ const NFTThumbnail = styled.Image<{ isFirst: boolean; isLast: boolean }>`
...
+  flex-direction: row;
+  gap: 15px;
+  align-items: flex-start;
+  padding-top: 16px;
+`
+
+const isAsset = (item: CarouselPageEntry): item is Asset => !!(item as Asset).id
+
+const isUnknownTokens = (item: CarouselPageEntry): item is UnknownTokensEntry =>
+  !!(item as UnknownTokensEntry).numberOfUnknownTokens
```
[the remark this answers is not in the document]
Comment (nop33) [reply src/components/AddressesTokensList.tsx:211]: Very good catch 💪 Thank you!!

**Comment by nop33** [comment]
I addressed all issues and pushed! Thanks for the great review!

**Review by mvaivre — APPROVED** [review]
Very good :)

**Pull request merged**